### PR TITLE
fix: support special chars in assert_contains expected expression

### PR DESCRIPTION
**Pull request by jbedard**
See https://github.com/aspect-build/rules_js/issues/1045

### Type of change

- Bug fix (change which fixes an issue)

### Test plan

- New test cases added

### CODE REVIEW

File: lib/testing.bzl [hunk 1]
```diff
@@ -22,24 +22,31 @@ def assert_contains(name, actual, expected, size = None, timeout = None, **kwarg
     """
 
     test_sh = "_{}_test.sh".format(name)
+    expression_txt = "_{}_expression.txt".format(name)
```
Comment (jbedard) [inline lib/testing.bzl:25]: Any internal file naming preferences or suggestions?

Comment (kormide) [reply lib/testing.bzl:25]: `expected`? Does it need the `.txt` extension?

**Review by kormide — APPROVED** [review]

File: lib/testing.bzl (hunk 1 above)
Comment (jbedard) [reply lib/testing.bzl:25]: There's really no restrictions on the name as long as it's unique to the target. I figured an extension is nice when browsing bazel-bin, but `.txt` feels odd 🤷 

`expected` sgtm though 👍

Comment (gregmagolan) [reply lib/testing.bzl]: +1 to no extension

Comment (jbedard) [reply lib/testing.bzl]: Note that `assert_outputs` also uses .txt 🤷

**Review by gregmagolan — APPROVED** [review]

File: lib/testing.bzl (hunk 1 above)
Comment (gregmagolan) [reply lib/testing.bzl]: Its a nit either way since it is not user facing

Comment (jbedard) [reply lib/testing.bzl]: I'm going to stick with .txt since the others in this file use it. We can change them all in the future if anyone wants to.

**Pull request merged**